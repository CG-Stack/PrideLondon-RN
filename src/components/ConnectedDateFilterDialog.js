// @flow
import { connect } from "react-redux";
import type { Connector } from "react-redux";
import { updateEventFilters } from "../actions/event-filters";
import type { DateOrDateRange } from "../data/date-time";
import { selectFilteredEvents } from "../selectors/events";
import { selectDateFilter } from "../selectors/event-filters";
import Component from "./DateRangePickerDialog";
import text from "../constants/text";

type OwnProps = {
  onApply: () => void,
  onCancel: () => void,
  visible: boolean
};

type Props = {
  applyButtonText: string,
  dateRange: ?DateOrDateRange,
  onChange: (?DateOrDateRange) => void
} & OwnProps;

const mapStateToProps = state => ({
  applyButtonText: text.filterPickerApply(selectFilteredEvents(state).length),
  dateRange: selectDateFilter(state)
});

const mapDispatchToProps = {
  onChange: date => updateEventFilters({ date })
};

const connector: Connector<OwnProps, Props> = connect(
  mapStateToProps,
  mapDispatchToProps
);

export default connector(Component);
