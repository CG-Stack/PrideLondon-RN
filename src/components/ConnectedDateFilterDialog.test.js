import React from "react";
import configureStore from "redux-mock-store";
import { shallow } from "enzyme";
import ConnectedDateFilterDialog from "./ConnectedDateFilterDialog";

const initialState = {
  events: {
    entries: [],
    assets: [],
    loading: true,
    refreshing: false
  },
  eventFilters: {
    selectedFilters: {
      categories: new Set(),
      date: null,
      time: new Set()
    },
    stagedFilters: {
      categories: new Set(),
      date: null,
      time: new Set()
    }
  }
};
const mockStore = configureStore();

describe("ConnectedDateFilterDialog", () => {
  it("renders connector", () => {
    const store = mockStore(initialState);
    const output = shallow(
      <ConnectedDateFilterDialog
        store={store}
        onApply={() => {}}
        onCancel={() => {}}
        visible
      />
    );
    expect(output).toMatchSnapshot();
  });

  it("renders component", () => {
    const store = mockStore(initialState);
    const output = shallow(
      <ConnectedDateFilterDialog
        store={store}
        onApply={() => {}}
        onCancel={() => {}}
        visible
      />
    );
    expect(output.dive()).toMatchSnapshot();
  });
});
