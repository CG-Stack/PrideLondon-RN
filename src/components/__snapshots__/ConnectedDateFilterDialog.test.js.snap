// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedDateFilterDialog renders component 1`] = `
<Dialog
  applyButtonText="No events"
  headerRight={
    <Touchable
      accessibilityComponentType="button"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={undefined}
      delayLongPress={undefined}
      delayPressIn={50}
      delayPressOut={undefined}
      disabled={undefined}
      hitSlop={undefined}
      onLongPress={undefined}
      onPress={[Function]}
      onPressIn={undefined}
      onPressOut={undefined}
      pressRetentionOffset={undefined}
      style={undefined}
    >
      <Text
        markdown={false}
        style={Object {}}
        type="text"
      >
        Clear
      </Text>
    </Touchable>
  }
  onApply={[Function]}
  onCancel={[Function]}
  title="Select dates"
  visible={true}
>
  <DateRangePicker
    dateRange={null}
    onChange={[Function]}
  />
</Dialog>
`;

exports[`ConnectedDateFilterDialog renders connector 1`] = `
<DateRangePickerDialog
  applyButtonText="No events"
  dateRange={null}
  onApply={[Function]}
  onCancel={[Function]}
  onChange={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  visible={true}
/>
`;
